### ISSUE TITLE
Sodium changes fluid rendering compared to vanilla

### ISSUE THREAD

**Issue by coderbot16** [jellysquid3/Sodium#609]
A water flow rendered by the current Sodium next branch:

![2021-03-26_23.56.16.png](https://user-images.githubusercontent.com/22420959/112709445-ef7a8000-8e76-11eb-9c91-dc89adb39723.png)

The same water flow, rendered in vanilla:

![2021-03-26_23.59.03.png](https://user-images.githubusercontent.com/22420959/112709447-f86b5180-8e76-11eb-84c3-5892a5756991.png)

I'm not sure if this is intended behavior or not.

**Comment by LucilleTea** [comment]
This is #38 - lack of translucency sorting - right?

**Comment by coderbot16** [comment]
That's what I originally thought, however if you look closely at the top of the water flow, you can see that the geometry of the diagonal faces is quite different (a flat face instead of a "V" face). It seems like this is separate from the previous translucency sorting issues.

**Comment by coderbot16** [comment]
It's possible that this is actually related to the fact that vanilla is generating degenerate non-coplanar quads here, and since Sodium changes the orientation of vertices in quads due to lighting, it might be changing the appearance of the quads too.

**Comment by coderbot16** [comment]
A non-coplanar quad will render differently depending on how it is converted to triangles, and changing the orientation of the vertices changes how it is converted to triangles:

![ridge_valley.gif](https://user-images.githubusercontent.com/22420959/112709779-729cd580-8e79-11eb-95c5-3441161956e4.gif)

**Comment by mscatloaf** [comment]
Not sure if this is the correct place to post this, but I have had some other weird issues with water that seem similar to what your are having:
![2021-05-20_16 06 17](https://user-images.githubusercontent.com/80300628/119061826-1e4e3780-b98a-11eb-9e6b-2ba17141dd26.png)
![2021-05-20_16 05 34](https://user-images.githubusercontent.com/80300628/119061821-1d1d0a80-b98a-11eb-84f1-654c5e6d141d.png)
![2021-05-20_16 05 43](https://user-images.githubusercontent.com/80300628/119061823-1db5a100-b98a-11eb-85ba-5b958096d1de.png)
![2021-05-20_16 05 52](https://user-images.githubusercontent.com/80300628/119061824-1db5a100-b98a-11eb-8dca-9de6ad8d160d.png)
![2021-05-20_16 06 00](https://user-images.githubusercontent.com/80300628/119061825-1e4e3780-b98a-11eb-913f-88a36e4afb7f.png)

**Comment by LucilleTea** [comment]
@mscatloaf The problem you are showing here is #38. This issue is about the particular shape of the water flow, not the translucency problems when they overlap.

**Comment by MeeniMc** [comment]
This issue is still valid in 0.4.10. The difference is really hard to see though.

**Comment by EnterGerald** [comment]
I've got similar issues with my water as well. However, I've got an issue where water makes ice transparent for whatever reason.

![2023-06-21_16 41 04](https://github.com/CaffeineMC/sodium-fabric/assets/137363659/3474f541-f81a-4b1a-a0af-649b9f328712)
![2023-06-21_16 40 54](https://github.com/CaffeineMC/sodium-fabric/assets/137363659/30944d9c-7f43-40a8-be0c-14301869ed84)
![2023-06-21_16 40 17](https://github.com/CaffeineMC/sodium-fabric/assets/137363659/e26b7eba-4d3e-4ac7-a08c-837030a0ac65)

**Comment by astridellensdottir** [comment]
this is a duplicate of #38, it’s a translucency sorting issue. the fix is non-trivial, but the dev team is working on it.

**Comment by MeeniMc** [comment]
The original report is not #38, but the followup images are.

**Comment by douira** [comment]
As coderbot mentioned, the orientation of the crease is changed. I've discovered that the orientation of the crease is differs between the inside-facing and outside-facing quad of such a piece of water. I've attached two screenshots from within and outside of water with the crease highlighted.
![2023-10-10_01 19 49](https://github.com/CaffeineMC/sodium-fabric/assets/17120022/eb9d5028-bf28-4048-a8d9-a34a97e14661)
![2023-10-10_01 20 03](https://github.com/CaffeineMC/sodium-fabric/assets/17120022/dc37774f-6a0a-4323-bb7b-367ba5f33e78)
